test/e2e/framework.go

```diff
@@ -38,13 +38,31 @@ type Framework struct {
 
 	Namespace *api.Namespace
 	Client    *client.Client
+
+	nsCreateFunc func(string, *client.Client) (*api.Namespace, error)
 }
 
 // NewFramework makes a new framework and sets up a BeforeEach/AfterEach for
 // you (you can write additional before/after each functions).
 func NewFramework(baseName string) *Framework {
 	f := &Framework{
-		BaseName: baseName,
+		BaseName:     baseName,
+		nsCreateFunc: createTestingNS,
+	}
+
+	BeforeEach(f.beforeEach)
+	AfterEach(f.afterEach)
+
+	return f
+}
+
+// NewCustomFramework makes a new framework and sets up a BeforeEach/AfterEach
+// for you. It also allows you to customize namespace creation by supplying a
+// custom function.
+func NewCustomFramework(baseName string, nsCreateFunc func(string, *client.Client) (*api.Namespace, error)) *Framework {
```

Comment:
somewhat minor nit, but the name "NewCustomFramework" doesn't really describe well how it's different from NewFramework. In particular, what's "Custom"? The name implies nothing about the namespace creation function.

of course, I'm also having trouble trying to think of a much better name to recommend that isn't unnecessarily verbose.